crates/client/gateway/src/server/router.rs

```diff
@@ -3,53 +3,97 @@ use std::{convert::Infallible, sync::Arc};
+        (path, _, false) if path.starts_with("feeder/") => Ok(service_unavailable_response("Feeder")),
+        _ => {
+            log::warn!("Main router received invalid request: {path}");
+            Ok(not_found_response())
+        }
     }
 }
 
 // Router for requests related to feeder_gateway
-async fn feeder_gateway_router(req: Request<Body>, backend: Arc<MadaraBackend>) -> Result<Response<Body>, Infallible> {
-    match (req.method(), req.uri().path()) {
-        (&Method::GET, "/feeder_gateway/get_block") => {
+async fn feeder_gateway_router(
+    req: Request<Body>,
+    path: &str,
+    backend: Arc<MadaraBackend>,
+    eth_core_contract_address: H160,
+    eth_gps_statement_verifier: H160,
+    public_key: Felt,
+) -> Result<Response<Body>, Infallible> {
+    match (req.method(), path) {
+        (&Method::GET, "feeder_gateway/get_block") => {
             Ok(handle_get_block(req, backend).await.unwrap_or_else(Into::into))
         }
-        (&Method::GET, "/feeder_gateway/get_state_update") => {
+        (&Method::GET, "feeder_gateway/get_signature") => {
+            Ok(handle_get_signature(req, backend).await.unwrap_or_else(Into::into))
+        }
+        (&Method::GET, "feeder_gateway/get_state_update") => {
             Ok(handle_get_state_update(req, backend).await.unwrap_or_else(Into::into))
         }
-        (&Method::GET, "/feeder_gateway/get_class_by_hash") => {
+        (&Method::GET, "feeder_gateway/get_class_by_hash") => {
             Ok(handle_get_class_by_hash(req, backend).await.unwrap_or_else(Into::into))
         }
-        _ => Ok(not_found_response()),
+        (&Method::GET, "feeder_gateway/get_compiled_class_by_class_hash") => {
+            Ok(handle_get_compiled_class_by_class_hash(req, backend).await.unwrap_or_else(Into::into))
+        }
+        (&Method::GET, "feeder_gateway/get_contract_addresses") => {
+            Ok(handle_get_contract_addresses(eth_core_contract_address, eth_gps_statement_verifier)
+                .await
+                .unwrap_or_else(Into::into))
+        }
+        (&Method::GET, "feeder_gateway/get_public_key") => {
+            Ok(handle_get_publick_key(public_key).await.unwrap_or_else(Into::into))
+        }
+        _ => {
+            log::warn!("Feeder gateway received invalid request: {path}");
+            Ok(not_found_response())
+        }
     }
 }
 
 // Router for requests related to feeder
 async fn gateway_router(
     req: Request<Body>,
+    path: &str,
     add_transaction_provider: Arc<dyn AddTransactionProvider>,
 ) -> Result<Response<Body>, Infallible> {
     match (req.method(), req.uri().path()) {
-        (&Method::POST, "/feeder/add_transaction") => {
+        (&Method::POST, "feeder/add_transaction") => {
             Ok(handle_add_transaction(req, add_transaction_provider).await.unwrap_or_else(Into::into))
         }
-        _ => Ok(not_found_response()),
+        _ => {
+            log::warn!("Gateway received invalid request: {path}");
```

Comment:
it is not a good idea to display a warning every time a client makes a bad request

Reply:
Will move it to debug with a custom target. Was handy during debugging but I see how this could be an issue.